http-api/src/lib.rs

```diff
@@ -82,7 +97,65 @@ impl Context {
         Ok(())
     }
 
-    /// Populate a map between ssh fingerprints and their peer identities
+    fn cleanup_sessions(&self, map: &mut HashMap<String, AuthState>) -> Result<(), Error> {
+        let mut to_remove: Vec<String> = Vec::new();
+        let current_time = self.now()?;
+
+        for (key, value) in map.iter() {
+            match value {
+                AuthState::Authorized(auth) => {
+                    if let Some(exp_time) = auth.expiration_time {
+                        if exp_time < current_time {
+                            to_remove.push(key.clone());
+                        }
+                    }
+                }
+                AuthState::Unauthorized(auth) => {
+                    if auth.expiration_time < current_time {
```

Comment:
Same here?

Reply:
In this case we are doing the same like earlier, I'll look into if we can avoid pushing session ids into an array over which we than have to iterate again, doing it a little bit more efficient.